mysite/ct/views.py

```diff
@@ -155,8 +155,8 @@ def fsm_redirect(self, request, eventName=None, defaultURL=None,
         # now we check here whether event is actually from path matching
         # this node
         referer = request.META.get('HTTP_REFERER', '')
-        referer = referer[referer.find('/ct/'):]
-        if request.method == 'POST' and referer != self.fsmStack.state.path \
+        origin = request.META.get('HTTP_ORIGIN', '')
```

Comment:
This shows we need explicit test coverage that checks how the code behaves when HTTP_ORIGIN is missing.  That should have been included in the original implementation.